### PR TITLE
Connect C++ nnz to Python

### PR DESCRIPTION
**Pull request by johnkerl**
Context: #431 and #410

### CODE REVIEW

**Review by eddelbuettel — COMMENTED** [review]
I don't speak Python well enough to be of real help here so I abstain.  Thanks for thinking of me though.

File: apis/python/src/tiledbsoma/util.py [hunk 1]
```diff
@@ -91,3 +91,25 @@ def uri_joinpath(base: str, path: str) -> str:
...
+            return []
+        step = ids.step
+        if step is None:
+            if ids.start <= ids.stop:
+                step = 1
+            else:
+                step = -1
+        stop = ids.stop + step
+        return list(range(ids.start, stop, step))
+    raise Exception(f"expected ids as slice or list; got {type(ids)}")
```
Comment (gspowley) [inline apis/python/src/tiledbsoma/util.py:115]: We also need to support passing ids to `set_dim_points` as an Arrow array.

File: libtiledbsoma/CMakeLists.txt [hunk 2]
```diff
@@ -46,6 +46,7 @@ option(SUPERBUILD "If true, perform a superbuild (builds all missing dependencie
 option(CMAKE_IDE "(Used for CLion builds). Disables superbuild and sets the EP install dir." OFF)
 option(FORCE_BUILD_TILEDB "Forces a local build of TileDB instead of searching system paths." ON)
 option(TILEDB_FORCE_ALL_DEPS "Forces a local build of TileDB instead of searching system paths." OFF)
+# option(DOWNLOAD_TILEDB_PREBUILT "If tiledb is being super built, this controls downloading prebuilt artifacts or building from source" ON)
```
Comment (gspowley) [inline libtiledbsoma/CMakeLists.txt:49]: Old comment from debugging session?

**Review by gspowley — CHANGES_REQUESTED** [review]

File: libtiledbsoma/CMakeLists.txt (hunk 2 above)
Comment (johnkerl) [reply libtiledbsoma/CMakeLists.txt:49]: derp

File: apis/python/src/tiledbsoma/util.py (hunk 1 above)
Comment (johnkerl) [reply apis/python/src/tiledbsoma/util.py:115]: @gspowley this PR has some bits of #400 in it and it should not; sorry about that -- will rework

**Review by gspowley — APPROVED** [review]

**Pull request merged**